crates/test-programs/src/bin/cli_args.rs

```diff
@@ -0,0 +1,13 @@
+fn main() {
+    let args = std::env::args().skip(1).collect::<Vec<_>>();
+    assert_eq!(
+        args,
+        [
+            "hello",
+            "this",
+            // "", // TODO: this should be here.
```

Comment:
This is a bug somewhere and I'm not sure what, but my guess is the `clap` processing of arguments or something like that. I need to bottom this out but it's unrelated to this PR in that this PR doesn't introduce a new failure mode, it only exposes an existing one.

Reply:
Ok I have now since fixed this. The underlying issue was present before this PR but it's only exposed in this PR so I lumped th efix in here. The problem was that `clap`'s parsing of `PathBuf` requires the value to be non-empty and we were modeling all of the arguments to wasm as `Vec<PathBuf>`. I've switched to `Vec<OsString>` to enable passing through empty arguments.